packages/blockchain-wallet-v4-frontend/src/data/middleware/webSocket/socketd/messageTypes.js

```diff
@@ -0,0 +1,19 @@
+export const header = message => message.entity === 'header' && message.header
+export const ethSentPending = message =>
+  message.transaction &&
+  message.address === message.transaction.from &&
+  message.transaction.state === 'pending'
+export const ethSentConfirmed = message =>
+  message.transaction &&
+  message.address === message.transaction.from &&
+  message.transaction.state === 'confirmed'
+export const ethReceivedPending = message =>
+  message.transaction &&
+  message.address === message.transaction.to &&
+  message.transaction.state === 'pending'
+export const ethReceivedConfirmed = message =>
```

Comment:
all of these ETH functions are 90% same checks.  Can we create a helper function that takes the message and cared about state that will return? Something like following perhaps

```
const checkMessage = (message, state) => {
  const { transaction, address } = message
  return transaction && address === transaction.from &&
  transaction.state === state
}

export const ethReceivedPending = message => checkMessage(message, 'pending')
```